Source/Core/AxisAlignedBoundingBox.js

```diff
@@ -191,6 +193,28 @@ define([
         return Intersect.INTERSECTING;
     };
 
+    /**
+     * If the given position is not already within the box, projects the given position onto the box
+     * @param {Cartesian3} position The position being projected onto this AxisAlignedBoundingBox.
+     * @param {Cartesian3} [result] The object onto which to store the result.
+     * @returns {Cartesian3} A projected version of the inputted position if it was not originally within the AxisAlignedBoundingBox.
+     */
+    AxisAlignedBoundingBox.prototype.clampToBounds = function(position, result) {
```

Comment:
I would require the `result` parameter.  We usually only make it optional for functions like `fromCartesian3`, `fromRectangle`, where you're creating a new instance

Reply:
Same for the other `clampToBounds` functions in the other bounding volume types